### PR TITLE
Improve options validation in Mix.Tasks.Compile.Elixir

### PR DESCRIPTION
**Pull request by whatyouhide**
On the road to #5388.

I'm on the fence on what to do with `:elixirc_options` here, since they're not set and changed here. Not sure where would be the best place to validate those 😕

### CODE REVIEW

File: lib/mix/lib/mix/tasks/compile.elixir.ex [hunk 1]
```diff
@@ -74,4 +76,10 @@ defmodule Mix.Tasks.Compile.Elixir do
     dest = Mix.Project.compile_path
     Mix.Compilers.Elixir.clean(manifest(), dest)
   end
+
+  defp assert_valid_elixirc_paths(paths) do
+    unless is_list(paths) do
+      Mix.raise ":elixirc_paths should be a list of paths, got: #{inspect(paths)}"
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/compile.elixir.ex]: Should we say ":elixirc_paths in project configuration should be...."?

**Comment by josevalim** [comment]
Regarding elixirc_options, I would validate them in the same place, even if they are used downstream. There is a chance Code.compiler_options/1 already validate them too.

File: lib/mix/lib/mix/tasks/compile.elixir.ex (hunk 1 above)
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/compile.elixir.ex]: I thought about that but if we get here I think users had to explicitly use `:elixirc_paths` so they likely know where it comes from?

**Comment by whatyouhide** [comment]
@josevalim since I merged #5472 should we go ahead and merge this one as well? The compiler options should be covered.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**